src/response.rs

```diff
@@ -113,6 +113,32 @@ impl Response {
     pub fn json<T: DeserializeOwned>(&mut self) -> ::Result<T> {
         serde_json::from_reader(self).map_err(::error::from)
     }
+
+    /// Turn a response into an error if the server returned an error.
+    ///
+    /// # Example
+    ///
+    /// ```rust,no_run
+    /// extern crate reqwest;
+    ///
+    /// fn main() {
+    ///     let resp = reqwest::get("http://127.0.0.1/")
+    ///         .expect("network error")
+    ///         .error_for_status()
+    ///         .expect("client or server error");
+    /// }
+    /// ```
+    pub fn error_for_status(self) -> ::Result<Self> {
+        match self.status().class() {
+            StatusClass::ClientError => {
+                Err(::error::client_error(self.url().clone(), *self.status()))
+            }
+            StatusClass::ServerError => {
+                Err(::error::server_error(self.url().clone(), *self.status()))
```

Comment:
I don't think so. At the very least, we'd need to duplicate the body of `Response::url`to get an owned version. But when I try that, I can't extract the url from the `hyper::client::response::Response` since that implements `Drop`.